pkg/provider-local/controller/infrastructure/actuator.go

```diff
@@ -82,13 +82,36 @@ func (a *actuator) Reconcile(ctx context.Context, _ logr.Logger, infrastructure
 	networkPolicyAllowToIstioIngressGateway := emptyNetworkPolicy("allow-to-istio-ingress-gateway", infrastructure.Namespace)
 	networkPolicyAllowToIstioIngressGateway.Spec = networkingv1.NetworkPolicySpec{
 		Egress: []networkingv1.NetworkPolicyEgressRule{{
-			To: []networkingv1.NetworkPolicyPeer{{
-				NamespaceSelector: &metav1.LabelSelector{MatchLabels: map[string]string{"kubernetes.io/metadata.name": "istio-ingress"}},
-				PodSelector: &metav1.LabelSelector{MatchLabels: map[string]string{
-					"app":   "istio-ingressgateway",
-					"istio": "ingressgateway",
-				}},
-			}},
+			To: []networkingv1.NetworkPolicyPeer{
+				{
+					NamespaceSelector: &metav1.LabelSelector{MatchLabels: map[string]string{"kubernetes.io/metadata.name": "istio-ingress"}},
+					PodSelector: &metav1.LabelSelector{MatchLabels: map[string]string{
+						"app":   "istio-ingressgateway",
+						"istio": "ingressgateway",
+					}},
+				},
+				{
+					NamespaceSelector: &metav1.LabelSelector{MatchLabels: map[string]string{"kubernetes.io/metadata.name": "istio-ingress--0"}},
+					PodSelector: &metav1.LabelSelector{MatchLabels: map[string]string{
+						"app":   "istio-ingressgateway",
+						"istio": "ingressgateway--zone--0",
+					}},
+				},
+				{
+					NamespaceSelector: &metav1.LabelSelector{MatchLabels: map[string]string{"kubernetes.io/metadata.name": "istio-ingress--1"}},
+					PodSelector: &metav1.LabelSelector{MatchLabels: map[string]string{
+						"app":   "istio-ingressgateway",
+						"istio": "ingressgateway--zone--1",
+					}},
+				},
+				{
+					NamespaceSelector: &metav1.LabelSelector{MatchLabels: map[string]string{"kubernetes.io/metadata.name": "istio-ingress--2"}},
+					PodSelector: &metav1.LabelSelector{MatchLabels: map[string]string{
+						"app":   "istio-ingressgateway",
+						"istio": "ingressgateway--zone--2",
+					}},
+				},
```

Comment:
Instead of hard-coding this and adding it always, can you use the zone information from the `Seed` and maybe some utilities from the `istio` package to get to the labels?

Reply:
Will be changed to a loop in the next commit. However, after offline discussion I will not use utility functions from the `istio` package to omit undesired import dependencies from provider-local to the botanist.